drivers/hmis/app/graphql/mutations/ac_hmis/update_referral_posting.rb

```diff
@@ -48,6 +49,19 @@ def resolve(id:, input:)
           end
           raise ActiveRecord::Rollback if errors.any?
 
+          # Similar to query_type.rb:project_can_accept_referral, but returns info about each conflicting enrollment, so the user can fix the errors.
+          personal_ids = posting.referral.household_members.map(&:client).pluck(:PersonalID)
+          # no need to check viewable_by on the enrollments, since we would have already raised 'access denied' above
+          conflicting_enrollments = posting.project.enrollments.open_including_wip.where(personal_id: personal_ids)
+
+          unless conflicting_enrollments.empty?
+            conflicting_enrollments.each do |e|
+              errors.add :base, :invalid, full_message: "#{e.client.full_name} already has an open enrollment in this project (entry date: #{e.entry_date}). Please exit the client if the enrollment is invalid or out-of-date, and otherwise deny the referral."
+            end
+
+            raise ActiveRecord::Rollback
```

Comment:
This is okay, but I also think this check doesn't _need_ to happen inside the transaction. It would be OK to check for conflicting enrollments first (if `posting_status_change == ['assigned_status', 'accepted_pending_status']`), and return immediately if there are errors due to the conflicting enrollments. Then we don't need to worry about rolling anything back.

I think that is generally safer, because we don't need to rely on the db transaction rollback behavior. The downside is that you only get the "first" error, for example if there are conflicting enrollments AND not enough units available, you wouldn't see the unit error until you resolved the conflicting enrollment errors.